src/pspm_compute_visual_angle.m

```diff
@@ -93,92 +89,75 @@
-        end
-        channels(str_chans) = str_chan_num;
-        work_chans = cell2mat(channels);
-        
-        if numel(work_chans) >= 1
             
-            % always use first found channel
+        % always use first found channel
+
+        gx = find(cellfun(@(x) strcmpi(gaze_x, x.header.chantype) & ...
+            strcmpi('mm', x.header.units), data),1);
+        gy = find(cellfun(@(x) strcmpi(gaze_y, x.header.chantype) & ...
+            strcmpi('mm', x.header.units), data),1);
             
-            gx = find(cellfun(@(x) strcmpi(gaze_x, x.header.chantype) & ...
-                strcmpi('mm', x.header.units), data),1);
-            gy = find(cellfun(@(x) strcmpi(gaze_y, x.header.chantype) & ...
-                strcmpi('mm', x.header.units), data),1);
+        if ~isempty(gx) && ~isempty(gy)
+
+            visual_angl_chans{p} = data{gx};
+            visual_angl_chans{p+1} = data{gy};
+
+            % get channel specific data
+            gx_d = data{gx}.data;
+            gy_d = data{gy}.data;
             
-            if ~isempty(gx) && ~isempty(gy)
-                
-                visual_angl_chans{p} = data{gx};
-                visual_angl_chans{p+1} = data{gy};
-                
-                % get channel specific data
-                gx_d = data{gx}.data;
-                gy_d = data{gy}.data;
-                gy_d = data{gy}.header.range(2)-gy_d;
-                
-                N = numel(gx_d);
-                if N~=numel(gy_d)
-                    warning('ID:invalid_input', 'length of data in gaze_x and gaze_y is not the same');
-                    return;
-                end;
-                
-                % move (0,0) into center of the screen
-                gx_d = gx_d - width/2;
-                gy_d = gy_d - height/2;
-                
-                %compute visual angle for gaze_x and gaze_y
-                s_x = gx_d; % x axis in spherical coordinates
-                s_y = distance * ones(numel(gx_d),1);% y axis in spherical coordinates, actually is the radial from participant to the screen
-                s_z = gy_d;  % z axis in spherical coordinates, actually is y axis on the screen
-                [azimuth, elevation, r]= cart2sph(s_x,s_y,s_z);% convert cartesian to spherical coordinates in radians, azimuth = longitude, elevation = latitude
-                
-                lat = rad2deg(elevation);% convert radians into degrees
-                lon = rad2deg(azimuth);
-                
-                %compute visual angle for range 
-                r_x = [-width/2,width/2,0,0]';
-                r_y = distance * ones(numel(r_x),1);
-                r_z = [0,0,-height/2,height/2]';
-                [x_range_sp, y_range_sp,r_range]= cart2sph(r_x,r_y,r_z);
-                x_range_sp = rad2deg(x_range_sp);
-                y_range_sp = rad2deg(y_range_sp);
+            % The convention is that the origin of the screen is in the bottom 

```

Comment:
I think we should mention this assumption in the function documentation, because other eye trackers may be using different conventions.

Reply:
Copy that